pyro/contrib/gp/likelihoods/binary.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import, division, print_function
+
+import torch.nn.functional as F
+
+import pyro
+import pyro.distributions as dist
+
+from .likelihood import Likelihood
+
+
+class Binary(Likelihood):
+    """
+    Implementation of Binary likelihood, which is used for binary classification.
+
+    :param callable response_function: A mapping to correct domain for Binary likelihood.
+        By default, we use `sigmoid` function.
+    """
+    def __init__(self, response_function=None):
+        super(Binary, self).__init__()
+        self.response_function = (response_function if response_function is not None
+                                  else F.sigmoid)
+
+    def forward(self, f, y=None):
+        f_res = self.response_function(f)
+        if y is None:
```

Comment:
@fehiepsi I'm surprised that this cannot be simplified to take advantage of broadcasting in `torch.distributions.Distribution.log_prob()`:
```py
def forward(self, f, y=None):
    f_res = self.response_function(f)
    return pyro.sample("y", dist.Bernoulli(f_res), obs=y)
```
Were you seeing errors in this simplified version? If so maybe we can fix those instead of working around them here.

Reply:
I got invalid log_prob shape, expected [], actually [2 , 3] for y_binary2D case.